src/remediations/fifi.js

```diff
@@ -428,6 +494,41 @@ function normalize (satellites, smart_management) {
     }));
 }
 
+async function prepareReceptorSatellites(satellites, account) {
+    const sourceInfo = await sources.getSourceInfo(_(satellites).map('id').filter().value());
+
+    _.forEach(satellites, satellite => {
+        satellite.source = _.get(sourceInfo, satellite.id, null);
+        satellite.receptor = getReceptor(satellite.source);
+    });
+
+    await P.map(satellites, async satellite => {
+        satellite.receptorStatus = await fetchReceptorStatus(satellite.receptor, account);
+    });
+}
+
+async function defineRHCSatellites(systems, org_id) {
+    const rhcSatellites = _(systems)
+    .groupBy((system) => `${system.satelliteId}+${system.satelliteOrgId}`)
+    .mapValues(systems => ({
+        id: systems[0].satelliteId,
+        org_id: systems[0].satelliteOrgId,
+        version: systems[0].satelliteVersion,
+        // unique by ansible host i.e. if there are two systems with the same ansible identifier then
+        // only pick on one of them as we wouldn't be able to tell them apart based on responses from Satellite
+        systems: _(systems).sortBy('id').uniqBy(generator.systemToHost).value()
+    })).values().value();
+
+    await P.map(rhcSatellites, async satellite => {
+        satellite.type = 'RHC-satellite';
+        satellite.source = await sources.getRHCConnections(satellite.id);
```

Comment:
You already fetched the id in `fetchSatRHCClientId`, no? Is it possible to reuse that info?